From 5405525961ea6f160c160a09ee21c756096af20d Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Sun, 10 Jan 2016 00:38:37 +0800
Subject: [PATCH 3633/5242] MLK-12203-2 ARM: imx: adjust loop number to delay
 ~20us for i.MX7D low power idle

commit  374c9810b0aaa8a031e4aae500a9b2d301c4cc5f from
https://source.codeaurora.org/external/imx/linux-imx.git

ARM does NOT execute one instruction every cycle, the bus bandwidth,
cache status etc. would impacts the instruction execution time, so we
can NOT just calculate the delay time by ARM frequency, this patch
adjusts loop number to get a ~20us delay, measured via GPIO pin.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/imx7d_low_power_idle.S |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-imx/imx7d_low_power_idle.S b/arch/arm/mach-imx/imx7d_low_power_idle.S
index 2cc088d..07460dd 100644
--- a/arch/arm/mach-imx/imx7d_low_power_idle.S
+++ b/arch/arm/mach-imx/imx7d_low_power_idle.S
@@ -278,8 +278,8 @@
 	ldr	r7, =0xf0000
 	str	r7, [r10]
 
-	/* assume ARM @ 1GHz, about delay 20us */
-	ldr	r7, =10000
+	/* delay 20us, measured by gpio */
+	ldr	r7, =20
 12:
 	subs	r7, r7, #0x1
 	bne	12b
@@ -296,7 +296,7 @@
 	ldr	r7, =0x0
 	str	r7, [r10]
 
-	ldr	r7, =10000
+	ldr	r7, =20
 13:
 	subs	r7, r7, #0x1
 	bne	13b
-- 
1.7.9.5

